### ISSUE TITLE
"End job" function in order to create multiple jobs from the same python script

### ISSUE THREAD

**Issue by cbalaf** [dessa-oss/atlas#119]
My use case is tracking every iteration of a hyperparameter optimization process. I've created a function that roughly looks like:
```python
def model_score(hyperparameters):
    xtrain, ytrain, xtest, ytest = get_data()
    model = create_model(hyperparameters)
    model.fit(xtrain, ytrain)
    score = model.evaluate(xtest, ytest)
    return score
```
I am then passing this function to my favorite optimization algorithm (like `skopt`'s gp_minimize, `hyperopt`'s fmin and so on) but I also want to log the inputs and outputs in the process:
```python
def model_score(hyperparameters):
    foundations.log_params(hyperparameters)
    xtrain, ytrain, xtest, ytest = get_data()
    model = create_model(hyperparameters)
    model.fit(xtrain, ytrain)
    score = model.evaluate(xtest, ytest)
    foundations.log_metric(score)
    return score
```

While this code doesn't raise any errors, it only logs the first parameter iteration as a "job" and assigns all the metric iterations to it. I'm wondering if it would be possible to have something like a `foundations.end_job()` function that I would call once I log my metrics so that the next time `log_parameters` is called, it creates a new job.

**Comment by mohammedri** [comment]
Hey @cbalaf , curious - can't you use the submit job method in the Python SDK to achieve this?
The SDK allows you to submit jobs programatically
https://docs.atlas.dessa.com/en/latest/sdk-reference/SDK/#job-submission

**Comment by cbalaf** [comment]
does the deployment object contain information about the job? If so, then yes, otherwise the issue is that I want the input to a subsequent job to depend on a previous one - which means that I would like to have output from a job returned to the program

**Comment by pippinlee** [comment]
It depends what information you need from the previous job.

If it's a metric from a previous job, can you try using [`.log_metric()`](https://docs.atlas.dessa.com/en/latest/sdk-reference/SDK/#log-metric) on the upstream job, which should give access to that resulting metric using the deployment object

**Comment by cbalaf** [comment]
oh, that's great! I guess my question then becomes on whether the upstream job needs to be a in a separate script that needs to be invoked or whether it can be a function defined in the same script

**Comment by pippinlee** [comment]
The way we organize our code is usually having a main file like `hyperparameter_serach.py` that acts as the driver, and kicks off all jobs for the search.

Here's a tutorial that hopefully clarifies usage of `foundations.submit` via random walk logic: https://docs.atlas.dessa.com/en/latest/tutorials/auction_price_regression_tutorial/#random-walk-search

It uses `get_metric` along with some basic logic to randomly walk through a search space. Let me know if this helps!

**Comment by shazraz** [comment]
hey @cbalaf , did the answer above help you get through the issue?

**Comment by cbalaf** [comment]
It helped seeing how I could do what I wanted within the current atlas logic. Didn't necessarily address the original reason why I opened this. In any case, this isn't a bug or a blocker, just my idea of an enhancement
